Add Documentation of the AppVeyor build process for NVDA forks

XLTechie opened the pull request:
### Link to issue number:

Follow-up of #16221 

### Summary of the issue:

The streamlined AppVeyor build process implemented in #16221, requires documentation in order to be useful.

### Description of user facing changes

None.

### Description of development approach

* Wrote a document in `projectDocs/dev/buildingNVDAOnAppVeyor.md`.
* Added references in `readme.md` and `contributing.md`, at @seanbudd's suggestion.
* Took the liberty of adding a missing lead heading in `buildingNVDA.md`.

### Testing strategy:

Checked all links to be working as of now.
Tested all commands.
I've been using variations of this process for three years.

### Known issues with pull request:

None.

### Code Review Checklist:
- [X] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing

Commit:
Added the missing lead heading to buildingNVDA.md

Commit:
Documented the NVDA fork build process on AppVeyor, in projectDocs/dev/buildingNVDAOnAppVeyor.md (#16293, followup of #16221)

lukaszgo1 commented on projectDocs/dev/buildingNVDAOnAppveyor.md:
```diff
@@ -0,0 +1,520 @@
+(More on the format and content later.)
+
+The most common placement for this file, is in the root directory of your repository.
+Unfortunately, since NV Access already uses Appveyor for official builds, it would be difficult to keep your build file in the root directory of your NVDA fork.This is because you will need to permanently change a few items in that file, and you wouldn't want that private build file "contaminating" your pull requests or diffs.
```
These sentences should preferably be each on its own line. If not there is a missing space:
```suggestion
Unfortunately, since NV Access already uses Appveyor for official builds, it would be difficult to keep your build file in the root directory of your NVDA fork. This is because you will need to permanently change a few items in that file, and you wouldn't want that private build file "contaminating" your pull requests or diffs.
```

XLTechie replied:
Thanks @lukaszgo1, missed that one. They are all supposed to be on their own lines.

XLTechie commented:
CC @Qchristensen
Don't know if you also review dev docs, but if so ...

seanbudd reviewed (commented):
generally looks good. I would suggest changing `Appveyor` to `AppVeyor` across this diff

XLTechie commented on projectDocs/dev/buildingNVDAOnAppveyor.md:
```diff
@@ -0,0 +1,570 @@
+    *(FixMe: more instructions may be needed here.)*
+4. Move to the "Select repository for your new project" heading, and then to the heading representing your GitHub username.
+5. Below that heading, you should find your GitHub repositories listed. Go to the one representing your NVDA fork.
+6. This part is less than perfectly accessible via the usual methods: you will need to use NVDA+numpad divide, to route the mouse to the repository name, and then numpad divide to click on it.
```
"Divide" falls outside of the backticks in your suggestion. I have modified the spacing and casing to reach your intended outcome.

```suggestion
6. This part is less than perfectly accessible via the usual methods: you will need to use `NVDA+NumpadDivide`, to route the mouse to the repository name, and then `NumpadDivide` to click on it.
```

seanbudd replied:
Key commands should be written in lowerCamelCase

https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/userGuideStandards.md#general-standards

The casing should be `numpadDivide`

XLTechie commented on projectDocs/dev/buildingNVDAOnAppveyor.md:
```diff
@@ -0,0 +1,570 @@
+
+That said, the best way to use this document, is to start at the beginning, and work your way through to the end, at least the first time you create an Appveyor build setup for NVDA.
+
+## Prerequisites
```
@seanbudd it is, though usually alternatives are given. Never the less it's a good idea, I have added a paragraph.

XLTechie commented:
@seanbudd I have addressed your review actions.
Also renamed the file accordingly, and updated the text of the commit itself in case you keep it.
GTG at my end.

seanbudd commented:
Just for future reference, it would have been better to keep the original commit. Commit messages don't matter much, and as it was a force push the PR has to be re-reviewed from scratch, rather than from the previous review

XLTechie commented:
> Just for future reference, it would have been better to keep the original commit. Commit messages don't matter much, and as it was a force push the PR has
> to be re-reviewed from scratch, rather than from the previous review

Sorry.

Commit:
Fix case

XLTechie commented:
@seanbudd Fixed.

Commit:
Fix a typo in Advanced topic: a build control branch

Commit:
Add a missing brace to the notifications section email template.
